x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/ItemSetBitSetTests.java

```diff
@@ -0,0 +1,213 @@
+        set2.set(200);
+        set2.set(65);
+
+        assertTrue(set2.isSubset(set1));
+        assertFalse(set1.isSubset(set2));
+        assertTrue(set1.isSubset(set1));
+
+        set2.set(0);
+        set2.set(5);
+        assertTrue(set2.isSubset(set1));
+        assertTrue(set1.isSubset(set2));
+
+        set2.set(99);
+        assertFalse(set2.isSubset(set1));
+        assertTrue(set1.isSubset(set2));
+
+        set1.set(999);
+        assertFalse(set1.isSubset(set2));
+        set2.set(999);
+        assertTrue(set1.isSubset(set2));
+        set2.set(2222);
+        assertTrue(set1.isSubset(set2));
+    }
+
+    public void testClone() {
+        ItemSetBitSet set1 = new ItemSetBitSet();
+        set1.set(0);
+        set1.set(3);
+        set1.set(200);
+        set1.set(5);
+        set1.set(65);
+
+        ItemSetBitSet set2 = (ItemSetBitSet) set1.clone();
+
+        assertTrue(set2.get(0));
+        assertFalse(set2.get(1));
+        assertFalse(set2.get(2));
+        assertTrue(set2.get(3));
+        assertTrue(set2.get(5));
+        assertFalse(set2.get(64));
+        assertTrue(set2.get(65));
+        assertTrue(set2.get(200));
+
+        set1.clear(200);
+        assertTrue(set2.get(200));
+
+        set1.set(42);
+        assertTrue(set1.get(42));
+        assertFalse(set2.get(42));
+    }
+
+    public void testReset() {
+        ItemSetBitSet set1 = new ItemSetBitSet();
+        set1.set(0);
+        set1.set(3);
+        set1.set(200);
+        set1.set(5);
+        set1.set(65);
+
+        ItemSetBitSet set2 = new ItemSetBitSet();
+        set2.reset(set1);
+        assertEquals(set1, set2);
+        assertEquals(set1.cardinality(), set2.cardinality());
+
+        assertTrue(set2.get(0));
+        assertFalse(set2.get(1));
+        assertFalse(set2.get(2));
+        assertTrue(set2.get(3));
+        assertTrue(set2.get(5));
+        assertFalse(set2.get(64));
+        assertTrue(set2.get(65));
+        assertTrue(set2.get(200));
+
+        set1.clear(200);
+        assertTrue(set2.get(200));
+
+        set1.set(42);
+        assertTrue(set1.get(42));
+        assertFalse(set2.get(42));
+
+        set2.set(99999999);
+        assertTrue(set2.get(99999999));
+
+        ItemSetBitSet set3 = new ItemSetBitSet();
+        set3.set(2);
+        set3.set(9);
+        set2.reset(set3);
+
+        assertEquals(set3, set2);
+        assertFalse(set2.get(99999999));
+    }
+
+    public void testHashCode() {
+        ItemSetBitSet set1 = new ItemSetBitSet();
+        set1.set(0);
+        set1.set(3);
+        set1.set(200);
+        set1.set(5);
+        set1.set(65);
+
+        ItemSetBitSet set2 = new ItemSetBitSet();
+        set2.reset(set1);
+
+        assertEquals(set1.hashCode(), set2.hashCode());
+        set2.set(99999999);
+        assertNotEquals(set1.hashCode(), set2.hashCode());
+        set2.clear(99999999);
+        assertEquals(set1.hashCode(), set2.hashCode());
+    }
+
+    public void testCompare() {
```

Comment:
Do you think it would be worth adding some focused `cardinality()` tests? It and superset are the two major changes from the original impl from what I can tell.